primefaces/src/main/resources/META-INF/resources/primefaces/autocomplete/autocomplete.js

```diff
@@ -609,8 +605,8 @@ PrimeFaces.widget.AutoComplete = PrimeFaces.widget.BaseWidget.extend({
                 var item = $(this);
 
                 if (!item.hasClass('ui-state-highlight')) {
-                    $this.items.filter('.ui-state-highlight').removeClass('ui-state-highlight');
-                    item.addClass('ui-state-highlight');
+                    $this.items.filter('.ui-state-highlight').removeClass('ui-state-highlight').attr('aria-selected', false);
+                    $this.highlightItem(item, true);
```

Comment:
This was the bug I think when unselected all highlighted items it should remove the selected as well